client/src/views/Viewer.vue

```diff
@@ -422,6 +422,11 @@ export default {
     /* END TODO */
     async loadDataset(datasetId) {
       var { data: dataset } = await this.girderRest.get(`folder/${datasetId}`);
+
+      if (!dataset) {
+        throw "Could not fetch dataset!";
```

Comment:
I've actually never seen anyone do this.  I found an old stackoverflow thread that says this behaves unexpectedly in IE and firefox, but that was in 2012.  My gut says `throw new Error()` should always be used, but maybe you know something I don't.